lite2/verifier.go

```diff
@@ -31,34 +40,34 @@ func Verify(
 	}
 
 	// Ensure last header can still be trusted.
-	expirationTime := h1.Time.Add(trustingPeriod)
-	if !expirationTime.After(now) {
-		return ErrOldHeaderExpired{expirationTime, now}
+	if HeaderExpired(h1, trustingPeriod, now) {
+		return ErrOldHeaderExpired{h1.Time.Add(trustingPeriod), now}
 	}
 
 	if err := verifyNewHeaderAndVals(chainID, h2, h2Vals, h1, now); err != nil {
-		return err
+		return ErrInvalidNewHeader{err}
 	}
 
 	if h2.Height == h1.Height+1 {
 		if !bytes.Equal(h2.ValidatorsHash, h1NextVals.Hash()) {
-			return errors.Errorf("expected old header validators (%X) to match those from new header (%X)",
+			err := errors.Errorf("expected old header next validators (%X) to match those from new header (%X)",
 				h1NextVals.Hash(),
 				h2.ValidatorsHash,
 			)
+			return ErrNewValSetCantBeTrusted{err}
```

Comment:
Shouldn't we return a different error here?
```suggestion
			return ErrInvalidNewHeader{err}
```
I think the caller deals with this by checking if these are sequential headers and returning an error (vs for example continuing bisection in the skipping header case). But I don't see an advantage of returning the `ErrNewValSetCantBeTrusted` error type. Or maybe just add a comment.

Reply:
Removed all the new errors except ErrNewValSetCantBeTrusted